apprentice_teacher.rb

```diff
@@ -1,50 +1,17 @@
+require_relative 'dbc_module'
+require_relative 'student'
+require_relative 'teacher_module'
+require_relative 'senior_teacher'
+
 class ApprenticeTeacher
-  attr_reader :age, :salary, :phase, :target_raise
-  attr_accessor :name
+  include DbcMod
+  include TeacherMod
+  attr_reader :salary, :target_raise
 
   def initialize(options={})
-    @age = options.fetch(:age, 0)
-    @name = options.fetch(:name, "")
-    @target_raise = 800
+    super(options)
     @phase = 3
-  end
-
```

Comment:
So in this case, the common state and behavior between the ApprenticeTeacher and SeniorTeacher should be extracted into a parent class named Teacher.

```ruby
class ApprenticeTeacher < Teacher
end

class SeniorTeacher < Teacher
end
```